data-prepper-plugins/aggregate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/TailSamplerAggregateAction.java

```diff
@@ -70,7 +71,12 @@ public AggregateActionResponse handleEvent(final Event event, final AggregateAct
     @Override
     public AggregateActionOutput concludeGroup(final AggregateActionInput aggregateActionInput) {
         GroupState groupState = aggregateActionInput.getGroupState();
-        return new AggregateActionOutput((List)groupState.getOrDefault(EVENTS_KEY, List.of()));
+        Random randomNum = new Random();
+        int randomInt = randomNum.nextInt(100);
+        if (((groupState.containsKey(ERROR_STATUS_KEY) && (Boolean)groupState.get(ERROR_STATUS_KEY) == true)) || (randomInt < percent)) {
```

Comment:
You are getting an int and then comparing to a double. I think this is somewhat minor, but it looked wrong to me at first. Typically, when doubles are used for percentages they are in the range of [0.0-1.0]. I did check and see that we use a double.

By the way, we might want to change the configuration `percent` to be a `BigDecimal` to avoid any issues with floating point numbers.

Reply:
No need to be concerned about minor representation issues of floating points. Actually, I can make the percent as Integer. It doesn't have to be double. I thought using double will provide flexibility to have more granular percentages, but probably not necessary